Sources/BraintreeCore/BTPaymentInsightsClient.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+///  Use `BTPaymentInsightsClient` to optimize your checkout experience by prioritizing the customer’s preferred payment methods in your UI.
+///  By customizing each customer’s checkout experience, you can improve conversion, increase sales/repeat buys and boost user retention/loyalty.
+///  - Note: This feature is in beta. It's public API may change in future releases.
+@objcMembers class BTPaymentInsightsClient: NSObject {
+    
+    private let apiClient: BTAPIClient
+    
+    /// Creates a `BTPaymentInsightsClient`
+    /// - Parameter apiClient: A `BTAPIClient` instance.
+    /// - Note: This features only works with a client token.
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+    }
+    
+    /// This method confirmes if the customer is a user of PayPal services using their email and phone number.
+    /// - Parameters:
+    ///   - email: The buyer's email address
+    ///   - phone: The buyer's phone number
+    /// - Returns: A `BTPaymentInsightsResult` instance
+    /// - Note: This feature is in beta. It's public API may change in future releases.
+    public func getRecommendedPaymentMethods(email: String, phone: String) async throws -> BTPaymentInsightsResult {
```

Comment:
Apparently we need to collect the phone's "countryCode" & "nationalNumber" separately. We should probably move to a `BTPaymentInsightsRequest` object as the input param instead

cc: @MaxHastingsPP @saperi22

Reply:
Done!